app/main.py

```diff
@@ -1,3 +1,57 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int or float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km!r})"
+
+    def __add__(self, other: int or "Distance") -> "Distance":
+        if isinstance(other, (int, float)):
+            return Distance(
+                self.km + other
+            )
+        return Distance(
+            self.km + other.km
+        )
+
+    def __iadd__(self, other: int or "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km += other
+        return self
```

Comment:
Perform type checking before conducting this and other similar operations as well

Reply:
according to tests method __ mul __ and __ truediv __ cannot receive Distance. So is there any need to check?